src/wasm.h

```diff
@@ -1688,11 +1688,13 @@ class Table : public Importable {
   static const Index kMaxSize = Index(-1);
 
   struct Segment {
+    Name table;
```

Comment:
Why do segments need to refer to the table that owns them? Shouldn't the table be available (or easily made available) to any code the uses a `Segment`? I think it would be simpler not to introduce a conceptual reference cycle here (table owns segments, segments refer to table).

Reply:
You're absolutely right, I totally forgot about the way we read or parse elem sections. I removed this.